### PR TITLE
update-profile

### PR DESCRIPTION
**Pull request by congvuong951999**
## Checklist
- [ ] This is bug/feature (delete incorrect one)
- [ ] Title including ticket number and title

## What this PR do?
Explain clearly and briefly what this PR does. Make a meaningful sentence instead of "this PR solve ticket #xxx"

## Context/Background
More information on the issue that you want to provide, related issues and PRs you have to write information to ask orally or in informal place too.

## Screenshots / Demo & Instruction for testing (if applicable)
image

- link to demo
- instruction on how to test this

## References
- Blog posts / web pages relevant to this issue (which you get your solution / idea from and it may help others to understand the issue quickly)
- Do we need update our knowledge base / project wiki?

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only:[:index, :edit, :update, :destroy, :following, :followers]
```
Comment (rikai-hai) [inline app/controllers/users_controller.rb]: controller có vài thằng mà đâu ra lắm vậy

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
...
+  before_action :correct_user, only: [:edit, :update]
```
Comment (rikai-hai) [inline app/controllers/users_controller.rb]: cách xún dưới 1 line

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -16,11 +18,39 @@ def create
...
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = "Please log in."
+      redirect_to login_url
+    end
+  end
+
+  # Confirms the correct user.s
+  def correct_user
```
Comment (rikai-hai) [inline app/controllers/users_controller.rb:58]: dùng lun cho cả edit, update, show

File: app/views/users/edit.slim [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        = f.text_field :name, class:'form-control'
+        = f.label :email
+        = f.email_field :email, class:'form-control'
+        = f.label :address, t("users.new.addres")
+        = f.text_field :address , class:'form-control'
+        = f.label :password
+        = f.password_field :password, class:'form-control'
+        = f.label :password_confirmation, "Confirmation"
+        = f.password_field :password_confirmation, class:'form-control'
+        = f.submit"Save changes",class:"btn btn-primary btn_submit"
```
Comment (rikai-hai) [inline app/views/users/edit.slim]: endline

**Comment by congvuong951999** [comment]
image

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -16,11 +20,39 @@ def create
       render :new
     end
   end
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
```
Comment (rikai-hai) [inline app/controllers/users_controller.rb]: line này chạy mấy lần ? a ns dùng lại correct user mà

**Pull request merged**